### DISCUSSION TITLE
Problemas con la conexión SQL

### DISCUSSION THREAD

**Discussion by AstorArico72** [AstorArico72/Proyecto-Transversal-1#4] [General]
Intenté subir datos a una base de datos SQL vacía con el archivo de conexión que la profesora C. Sáez envió.
No parece funcionar, ¿Cómo arreglo éso?

En adición a éso, ya encontré qué usar en lugar de class.newInstance (), método obsoleto desde Java SE 9.

> La alternativa es class.getDeclaredConstructor ().newInstance ()

> Toda la línea sería, en nuestro caso: `Class.forName("org.mariadb.jdbc.Driver").getDeclaredConstructor ().newInstance();`

**Comment by AstorArico72** [comment]
Lo único que hice es cambiar algunas cosas en la solicitud SQL, corregir el newInstance obsoleto, y ejecutarlo, pero mi base de datos sigue en blanco.

**Reply by PabloBarrionuevo** [reply]
Buenas tardes, al final xampp te funciono para hacer pruebas? de conexión por ejemplo. Tendríamos que organizar un zoom para ver ese tema, no se si ya lo explicaron por clase.

**Reply by AstorArico72** [reply]
Con XAMPP inicio el servidor web Apache y MariaDB, y puedo usar PHPMyAdmin para crear, ver, y editar las bases de datos que tenga.
No sé si ésto es normal o no, pero al ejecutar la aplicación gráfica de XAMPP, tengo que abrir el bash, cambiar al directorio donde está (/opt/lampp), y correr algo así como:
`$ sudo ./Manager-Linux-x64.run`
Por alguna razón no hay ninguna aplicación de escritorio, aunque no creo necesitarla.

**Reply by PabloBarrionuevo** [reply]
Si funciona genial, para que todos tengamos los mismos archivos y mismos métodos de trabajo.

**Reply by AstorArico72** [reply]
Para aclarar malentendidos, dije que **no uso la aplicación gráfica**, que no tiene mucha utilidad.
Lo que hago en el bash es ésto:
`$ cd /opt/lampp`
`$ sudo ./xampp start`

> Éso inicia el servidor web Apache y MariaDB juntos. También puedo usar `$ sudo ./xampp startapache` para iniciar el servidor web Apache, o `$ sudo ./xampp startmysql` para iniciar MariaDB.

Para detener los servidores, hago exactamente lo mismo, pero cambiando "start" por "stop".
No sé por qué, pero siempre tengo que hacer "sudo" para los comandos de XAMPP.

**Comment by PabloBarrionuevo** [comment]
No vi bien el archivo de conexión de la profesora, pero debería tener:
Una conexión de java.sql.Connection; y un registro al driver que vamos a usar para conectar; (mariadb o mysql)

Después talvez un método para conectar y otro para desconectar.

El de conectar puede ser algo así, (sin ver el tema de excepciones):
Class.forName(driver); //donde driver es la dirección de la clase que vas a registrar;
VariabledeLaConexion = (java.sql.Connection) java.sql.DriverManager.getConnection(url, usuario, contrasenia);
//con cast a Connection si hace falta
//y usando el método de getConnection de DriverManager, con los datos de conexión -> establece la conexión

Luego, puede haber otro método, o algo para hacer un Statement.
Y con un ResultSet obtener de ese Statement ejecutando executeQuery, los datos de la BD.
O excepciones que hay que controlar.

Por ultimo un cierre de la conexión. 
Cualquier cosa, programamos un zoom y lo vemos bien, o esperemos las clases que lo expliquen (si no lo explicaron aun)

**Reply by AstorArico72** [reply]
Tiene todo éso, apenas que corregí un método deprecado.
Puede que tenga que incluir MariaDB o algo así.

**Reply by PabloBarrionuevo** [reply]
Claro en librerías de tu proyecto J. tenes que pegar o agregar el archivo de mariadb. Te tiene que quedar:
proyecto ...
  |_ la carpeta src
  |_ la carpeta de librerias
    |_y aca el archivo de mariadb
Probe con los dos archivos el de mysql y el de mariadb, y los dos funcionan. 
Estos archivos los descargue de Archivos en el aula de Lab1, se llaman:
mariadb-java-client-2.3.0.jar
mysql-connector-java-5.1.46-bin.jar
Con esas versiones funciono bien.

**Reply by PabloBarrionuevo** [reply]
Dentro de la carpeta de librerías. (no salen los espaciados)

**Reply by AstorArico72** [reply]
¿Debe ser un .jar?
Ya encontré MariaDB en /usr/lib64/mariadb, pero no hay ningún JAR.
¿Tengo que comprimir toda la biblioteca en un JAR, y añadirla al proyecto?

**Reply by AstorArico72** [reply]
Ya logré establecer una conexión con mi base SQL, tuve que descargar el controlador de conexión de MariaDB.
Bloqueo éste hilo mañana.
